### PR TITLE
SLING-8079 throw PostConstructException for createModel in case

### PR DESCRIPTION
**Pull request by kwin**
PostConstruct method returned false

Prevent creation of exception object for simple adaptations

### CODE REVIEW

File: src/main/java/org/apache/sling/models/impl/ModelAdapterFactory.java [hunk 1]
```diff
@@ -265,7 +265,11 @@ private void clearDisposalCallbackRegistryQueue() {
     public <AdapterType> AdapterType getAdapter(Object adaptable, Class<AdapterType> type) {
         Result<AdapterType> result = internalCreateModel(adaptable, type);
         if (!result.wasSuccessful()) {
-            log.warn("Could not adapt to model", result.getThrowable());
+            if (result == Result.POST_CONSTRUCT_PREVENTED_MODEL_CONSTRUCTION) {
+                log.warn("Could not adapt to model as PostConstruct method returned false"); // do no construct runtime exception in this case
```
Comment (stefanseifert) [inline src/main/java/org/apache/sling/models/impl/ModelAdapterFactory.java:269]: should this be converted to debug message? if the postconstruct returned false, this is by intention and not something to be warned?

Comment (kwin) [reply src/main/java/org/apache/sling/models/impl/ModelAdapterFactory.java]: good point, lowered the level to debug in https://github.com/apache/sling-org-apache-sling-models-impl/pull/28/commits/e3afef7b4b175a01ebbf7f7c5bcf0b8607c541bc.

**Comment by kwin** [comment]
@stefanseifert Why did you close without merging?

**Comment by stefanseifert** [comment]
sorry, by accident

**Comment by kwin** [comment]
@stefanseifert Can we merge or is anything outstanding?

**Review by stefanseifert — APPROVED** [review]

**Pull request merged**